pkg/log/rotate.go

```diff
@@ -157,18 +164,77 @@ func (f *FileWithRotation) rotate() error {
 		return fmt.Errorf("failed to rotate file, error while moving the current file: %v", err)
 	}
 
+	if f.cfg.Compress {
+		go func() {
+			if err := f.compress(rotateFileName); err != nil {
+				rLog.WithError(err).Error("Failed to compress rotated log file")
+
+				return
+			}
+			// Clean file that was compressed.
+			if err := os.Remove(rotateFileName); err != nil {
+				rLog.WithError(err).Error("Failed to clean rotated log file after was compressed")
+
+				return
+			}
+		}()
+	}
+
 	if err := f.open(); err != nil {
 		return fmt.Errorf("failed to create new file after rotation, error: %v", err)
 	}
 
 	return nil
 }
 
+// compress will create a .gz archive for the file provided.
+func (f *FileWithRotation) compress(file string) error {
+	srcFile, err := disk.OpenFile(file, os.O_RDWR|os.O_CREATE, filePerm)
+
+	defer func() {
+		_ = srcFile.Close()
```

Comment:
What do you think of having a named return error so we can log it case the close fails? As we need to perform multiple closes I would use the `go.uber.org/multierr` to append the errors into a single one. 
```suggestion
		err = multierr.Append(err, srcFile.Close())
```

Reply:
+1 to log the error